src/Development/IDE/Core/Shake.hs

```diff
@@ -600,7 +609,7 @@ defineOnDisk act = addBuiltinRule noLint noIdentity $
                     pure $ RunResult ChangedNothing (fromMaybe "" current) (isJust current)
                   else do
                     (diags, mbHash) <- runAct
-                    updateFileDiagnostics file (Key key) extras $ map snd diags
+                    updateFileDiagnostics file (Key key) extras $ map (\(_,y,z) -> (y,z)) diags
```

Comment:
Probably better for another PR but this seems to hint at the fact that a record type would be a better choice for `FileDiagnostic` than a 3-tuple.